### ISSUE TITLE
`expo-cli@4.13.0` - app stuck on splash screen

### ISSUE THREAD

**Issue by thedoublejay** [expo/expo-cli#3989]
### Summary

When updating to `expo-cli@4.13.0`, our app is stuck on the splash screen. It is working with `expo-cli@4.12.12`

### Environment

 Expo CLI 4.13.0 environment info:
    System:
      OS: Windows 10 10.0.22000
    Binaries:
      Node: 16.13.0 - C:\Program Files\nodejs\node.EXE
      npm: 7.16.0 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version     2020.3.0.0 AI-203.7717.56.2031.7678000
    npmPackages:
      @expo/webpack-config: ^0.16.7 => 0.16.7
      expo: 43.0.2 => 43.0.2
      react: 17.0.1 => 17.0.1
      react-dom: 17.0.1 => 17.0.1
      react-native: 0.64.3 => 0.64.3
      react-native-web: 0.17.1 => 0.17.1
    Expo Workflow: managed

### Please specify your device/emulator/simulator platform, model and version

4.13.0

### Reproducible demo or steps to reproduce from a blank project

We just run our project, however we define `main` in package.json with `index.js`. Not sure if it matters but I saw on the changelog regarding this change - https://github.com/expo/expo-cli/pull/3973/files

**Comment by calitb** [comment]
oh, happening to us too.  It also seems to be affecting the EAS build process, so right now we can't make any builds using eas

**Comment by EvanBacon** [comment]
Cannot repro, please provide more info. Perhaps try updating your `metro.config.js` version.

**Comment by calitb** [comment]
on my side it was related to another issue. One of our dependencies was configured wrong (using a different version of react-native than the one used by latest expo)

**Comment by thedoublejay** [comment]
The issue for me comes with `@expo/metro-config@0.3.0`. It was working with `@expo/metro-config@0.2.8`. I'm still checking what might have caused the issue. There are no logs on console and it's just stucked on the splash screen.

@EvanBacon my `metro.config.js` is very minimal.

```
const { getDefaultConfig } = require('@expo/metro-config')
const defaultConfig = getDefaultConfig(__dirname)

defaultConfig.resolver.extraNodeModules.stream = require.resolve('stream-browserify')

module.exports = defaultConfig
```

**Comment by muathime** [comment]
I had the same problem so I did code review on my code on github and found out that I was importing `useState` wrongly. Seems to have slipped on an auto import event.

`import { useState } from 'react/cjs/react.development';`
So I corrected to
`import { useState } from "react";` 

In your case it might be different but I propose searching for any import from `....react.development` and correct that.

I resolved it this way and now it works great!!!
